shell/profiling/profiler.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PROFILING_PROFILER_H_
+#define FLUTTER_SHELL_PROFILING_PROFILER_H_
+
+#include <functional>
+#include <memory>
+
+#include "flutter/fml/task_runner.h"
+#include "flutter/fml/trace_event.h"
+
+namespace flutter {
+
+struct CpuUsageInfo {
+  uint32_t num_threads;
+  double total_cpu_usage;
+};
+
+// Return value is the total CPU time spent by all threads owned by the current
+// application. Sometimes the sampler can fail, hence the optional.
+using CpuUsage = std::function<std::optional<CpuUsageInfo>(void)>;
+
+class Profiler {
+ public:
+  Profiler(fml::RefPtr<fml::TaskRunner> profiler_task_runner,
+           CpuUsage cpu_usage,
+           int num_samples_per_sec);
+
+  void Start();
```

Comment:
`bool Start() const;` The return type because it would be easier for caller to check for errors and prefer const correctness always (go/fluttercpp#Use_of_const)

Reply:
done!